cibuildwheel/macos.py

```diff
@@ -122,39 +123,54 @@ def call(args, env=None, cwd=None, shell=False):
         delocated_wheel = glob('/tmp/delocated_wheel/*.whl')[0]
 
         if test_command:
-            # set up a virtual environment to install and test from, to make sure
-            # there are no dependencies that were pulled in at build time.
-            call(['pip', 'install', 'virtualenv'], env=env)
-            venv_dir = tempfile.mkdtemp()
-            call(['python', '-m', 'virtualenv', venv_dir], env=env)
-
-            virtualenv_env = env.copy()
-            virtualenv_env['PATH'] = os.pathsep.join([
-                os.path.join(venv_dir, 'bin'),
-                virtualenv_env['PATH'],
-            ])
-            # Fix some weird issue with the shebang of installed scripts
-            # See https://github.com/theacodes/nox/issues/44 and https://github.com/pypa/virtualenv/issues/620
-            virtualenv_env.pop('__PYVENV_LAUNCHER__', None)
-
-            # check that we are using the Python from the virtual environment
-            call(['which', 'python'], env=virtualenv_env)
-
-            # install the wheel
-            call(['pip', 'install', delocated_wheel + test_extras], env=virtualenv_env)
-
-            # test the wheel
-            if test_requires:
-                call(['pip', 'install'] + test_requires, env=virtualenv_env)
-
-            # run the tests from $HOME, with an absolute path in the command
-            # (this ensures that Python runs the tests against the installed wheel
-            # and not the repo code)
-            test_command_prepared = prepare_command(test_command, project=abs_project_dir)
-            call(test_command_prepared, cwd=os.environ['HOME'], env=virtualenv_env, shell=True)
-
-            # clean up
-            shutil.rmtree(venv_dir)
+            # Because of https://github.com/pypa/virtualenv/issues/1437, we can only test
+            # 32-bit mode using python 3 embedded venv module.
+            if 'intel' in config.identifier and config.version[0] == '3':
```

Comment:
maybe change this to checking wheel file name. Then if there is method to "repair" wheel after construction it will still work.